### ISSUE TITLE
Different types of Array Highlights

### ISSUE THREAD

**Issue by EilrahcF** [MusicTheorist/ArrayVisualizer#99]
This would certainly take time, but what if the program had the ability to mark certain elements in an array, say a different colour depending on their use, exactly like Timmo Bingman's program does?

Of course if this does actually get concidered there should be a way to toggle it on and off for those who don't want it, and it would certainly take an extremely long time to apply this to all the sorts in the program and also we would need to find out how to do this with the already colourful visual styles, but still, I think it could be an amazing idea.

Another plus would be that it would make teaching people about these algorithms more interesting due to a difference between different parts of the array, like for example teaching people about grailsort could be more interesting due to, say the scrolling buffer could be a different colour to the rest of the array, or with insertion sort, say, the sorted part could be a different colour to the rest of the array and so on.

In conclusion this would make the more complicated sorts easier to understand, along with some of the easier ones, but again, the only downside would be the time it would take to implement.
